### ISSUE TITLE
Usage · pirate/ArchiveBox Wiki

### ISSUE THREAD

**Issue by tg-z** [tg-z/web-clip#310]
Usage · pirate/ArchiveBox Wiki                                      

![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAPAgMAAABGuH3ZAAAAAXNSR0IArs4c6QAAAAlQTFRF/2YA////AAAALA0mBgAAAAFiS0dEAIgFHUgAAAAJcEhZcwAAEnMAABJzAYwiuQcAAAAHdElNRQfcCxUTCRv8+I/EAAAAMklEQVQI12NYtWrVCoapoaERDBMYGCSghAMLkBAQABIsDkCCkQHGAouBZeGKwXpBpgAAhtMRBKRIycIAAAAASUVORK5CYII=)
[![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAAAPBAMAAACB51W8AAAAAXNSR0IArs4c6QAAACRQTFRFAAEAAxx+JCYjYWNgho53kJKPy6Gio63Ys7Wysb3lyMnG/f/8AJKv4wAAAAFiS0dEAIgFHUgAAAAJcEhZcwAACxMAAAsTAQCanBgAAAAHdElNRQfcBhsNKDWiRqoQAAAA2UlEQVQoz42SoQ7CMBCGL10w7CWWEAwzNRiwOEzD5DwvgSGoJVUoBAZFSGeWQZPSeznabutYtiX709b0y3/X/wowVSFW0vZ4iSMOq2hAGVgux2d98dl15UFmQMVr4zHwAAsFsMXMAJdltKKzYVACgTVAbA2RucYdaLZfDtSJhJgGPMPvHRk5QwKpvWvXvi5NjZ3pkeP1hDQoiaxAb9eAb0gtqDneULMW7DmWplb8gI19rOiC3R4VkCgsYW5T539gPx7lhmITFH4So4G7qM1kpoBC5DiqYvLn+QGXavuS25r9yAAAAABJRU5ErkJggg==)
](https://github.com/pirate/ArchiveBox/commits/master.atom "https://github.com/pirate/ArchiveBox/commits/master.atom")

[Skip to content](#start-of-content)

    ![](https://github.githubassets.com/images/search-key-slash.svg)

*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    

*   No suggested jump to results

*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    
*   ![]()
    
    In this repository All GitHub ↵
    
    Jump to ↵
    
*   ![](https://github.githubassets.com/images/spinners/octocat-spinner-128.gif)
    

[Dashboard](/dashboard) [Pull requests](https://github.com/pulls?q=is%3Apr+is%3Aopen+author%3Atg-z+archived%3Afalse+sort%3Aupdated-desc) [Issues](https://github.com/issues?q=is%3Aissue+is%3Aopen+author%3Atg-z+archived%3Afalse+sort%3Aupdated-desc) [Trending](/trending) [Explore](/explore)[![](https://avatars3.githubusercontent.com/u/47585455?s=60&v=4)
 tg-z](https://github.com/tg-z) 

 Sign out 

[pirate](/pirate) / [ArchiveBox](/pirate/ArchiveBox)

[New repository](/new) [Import repository](/new/import) [New gist](https://gist.github.com/) [New organization](/organizations/new)

This repository

[New issue](/pirate/ArchiveBox/issues/new/choose) [New project](/pirate/ArchiveBox/projects/new)

 ![](https://avatars0.githubusercontent.com/u/47585455?s=40&v=4) [Signed in as **tg-z**](/tg-z)

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

lost

### Edit status

 

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

![](https://github.githubassets.com/images/icons/emoji/octocat.png)

  

Could not update your status, please try again.

80 remaining

#### Suggestions:

🌴

On vacation

🤒

Out sick

🏠

Working from home

🎯

Focusing

 Busy 

When others mention you, assign you, or request your review, GitHub will let them know that you have limited availability.

Clear status

Never

*   Never
    
    Keep this status until you clear your status or edit your status.
    

*   in 30 minutes
*   in 1 hour
*   in 4 hours
*   today
*   this week

Set status Clear status

[Your profile](/tg-z) [Your repositories](/tg-z?tab=repositories) [Your projects](/tg-z?tab=projects) [Your stars](/tg-z?tab=stars) [Your gists](https://gist.github.com/mine)

[Marketplace](/marketplace)

Feature preview

[Help](https://help.github.com) [Settings](/settings/profile)

 Sign out 

*   Sponsor
    
    ### Sponsor pirate/ArchiveBox
    
    ##### GitHub Sponsors
    
    [Learn more](/sponsors)
    
     [![](https://avatars1.githubusercontent.com/u/511499?s=100&v=4)](/pirate) 
    
    [Nick Sweeting](/pirate)
    
    [pirate](/pirate)
    
    [Sponsor](/sponsors/pirate)
    
    ##### External links
    
    ![](https://github.githubassets.com/images/modules/site/icons/funding_platforms/patreon.svg)
    
    [patreon.com/**theSquashSH**](https://patreon.com/theSquashSH)
    
    [https://paypal.me/NicholasSweeting](https://paypal.me/NicholasSweeting)
    
    [https://www.blockchain.com/eth/address/0x5D4c34D4a121Fe08d1dDB7969F07550f2dB9f471](https://www.blockchain.com/eth/address/0x5D4c34D4a121Fe08d1dDB7969F07550f2dB9f471)
    
    [https://www.blockchain.com/btc/address/1HuxXriPE2Bbnag3jJrqa3bkNHrs297dYH](https://www.blockchain.com/btc/address/1HuxXriPE2Bbnag3jJrqa3bkNHrs297dYH)
    
    Learn more about [funding links in repositories](https://help.github.com/articles/displaying-a-sponsor-button-in-your-repository).
    
    [Report abuse](/contact/report-abuse?report=pirate%2FArchiveBox+%28Repository+Funding+Links%29)
    
*         Watch   
    
    Notifications
    
    Not watching Be notified only when participating or @mentioned. Watch
    
    Releases only Be notified of new releases, and when participating or @mentioned. Unwatch releases
    
    Watching Be notified of all conversations. Unwatch
    
    Ignoring Never be notified. Stop ignoring
    
    [110](/pirate/ArchiveBox/watchers)
    
*       Unstar  [5.6k](/pirate/ArchiveBox/stargazers) 
    
        Star  [5.6k](/pirate/ArchiveBox/stargazers) 
    
*      Fork
    
    [315](/pirate/ArchiveBox/network/members)

[pirate](/pirate)/**[ArchiveBox](/pirate/ArchiveBox)**

Loading status checks…

==================================================================================

[Code  ](/pirate/ArchiveBox)   [  Issues 68  ](https://github.com/pirate/ArchiveBox/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)  [  Pull requests 4  ](https://github.com/pirate/ArchiveBox/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc) 

[Boards](#issue-sh-boards)

[Reports](#issue-sh-reports)

[Actions](/pirate/ArchiveBox/actions) [Wiki](/pirate/ArchiveBox/wiki) [Releases](/pirate/ArchiveBox/releases) More [Compare](/pirate/ArchiveBox/compare) [Dependencies](/pirate/ArchiveBox/network/dependencies) [Commits](/pirate/ArchiveBox/commits) [Branches](/pirate/ArchiveBox/branches) [Security](https://github.com/pirate/ArchiveBox/security/advisories) [Insights](https://github.com/pirate/ArchiveBox/pulse)

[Code  ](/pirate/ArchiveBox)   [ Issues 68  ](https://github.com/pirate/ArchiveBox/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)   [ Pull requests 4  ](https://github.com/pirate/ArchiveBox/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc)   [ Wiki  ](/pirate/ArchiveBox/wiki)  [ Security  ](/pirate/ArchiveBox/security/advisories) [ Pulse ](/pirate/ArchiveBox/pulse) [ Community ](/pirate/ArchiveBox/community) 

Usage
=====

[Jump to bottom](#wiki-pages-box)

Nick Sweeting edited this page on Mar 12 · [57 revisions](/pirate/ArchiveBox/wiki/Usage/_history)

▶️ _Make sure the dependencies are [fully installed](https://github.com/pirate/ArchiveBox/wiki/Install) before running any ArchiveBox commands._

**ArchiveBox API Reference:**

![](https://camo.githubusercontent.com/aaa8d77384b7c0fa835a1c1ab58eed23851efdb9/68747470733a2f2f692e696d6775722e636f6d2f61515a5a636b752e706e67)

*   [Overview](#Overview): Program structure and outline of basic archiving process.
*   [CLI Usage](#CLI-Usage): Docs and examples for the ArchiveBox command line interface.
*   [UI Usage](#UI-Usage): Docs and screenshots for the outputted HTML archive interface.
*   [Disk Layout](#Disk-Layout): Description of the archive folder structure and contents.

**Related:**

*   [Docker](/pirate/ArchiveBox/wiki/Docker): Learn about ArchiveBox usage with Docker and Docker Compose
*   [Configuration](/pirate/ArchiveBox/wiki/Configuration): Learn about the various archive method options
*   [Scheduled Archiving](/pirate/ArchiveBox/wiki/Scheduled-Archiving): Learn how to set up automatic daily archiving
*   [Publishing Your Archive](/pirate/ArchiveBox/wiki/Publishing-Your-Archive): Learn how to host your archive for others to access
*   [Troubleshooting](/pirate/ArchiveBox/wiki/Troubleshooting): Resources if you encounter any problems
*   [Screenshots](https://github.com/pirate/ArchiveBox#Screenshots): See what the CLI and outputted HTML look like

[](#overview)Overview
---------------------

The `./archive` binary is a shortcut to `bin/archivebox`. Piping RSS, JSON, [Netscape](https://msdn.microsoft.com/en-us/library/aa753582(v=vs.85).aspx), or TXT lists of links into the `./archive` command will add them to your archive folder, and create a locally-stored browsable archive for each new URL.

The archiver produces an [output folder](#Disk-Layout) `output/` containing `index.html`, `index.json`, and archived copies of all the sites organized by timestamp bookmarked. It's powered by [Chrome headless](https://developers.google.com/web/updates/2017/04/headless-chrome), good 'ol `wget`, and a few other common Unix tools.

[](#cli-usage)CLI Usage
-----------------------

![](https://camo.githubusercontent.com/c64243afbc2721c9e3a4883cc0be2fbf65416e6d/68747470733a2f2f692e696d6775722e636f6d2f626956664659722e706e67)

`./archive` refers to the executable shortcut in the root of the project, but you can also call ArchiveBox via `./bin/archivebox`. If you add `/path/to/ArchiveBox/bin` to your shell `$PATH` then you can call `archivebox` from anywhere on your system.

If you're using Docker, the CLI interface is similar but needs to be prefixed by `docker-compose exec ...` or `docker run ...`, for examples see the [Docker](/pirate/ArchiveBox/wiki/Docker) page.

*   [Run ArchiveBox with configuration options](#Run-ArchiveBox-with-configuration-options)
*   [Import a single URL or list of URLs via stdin](#Import-a-single-URL-or-list-of-URLs-via-stdin)
*   [Import list of links exported from browser or another service](#Import-list-of-links-exported-from-browser-or-another-service)
*   [Import list of URLs from a remote RSS feed or file](#Import-list-of-URLs-from-a-remote-RSS-feed-or-file)
*   [Import list of links from browser history](#Import-list-of-links-from-browser-history)

* * *

### [](#run-archivebox-with-configuration-options)Run ArchiveBox with configuration options

You can set environment variables in your shell profile, a config file, or by using the `env` command.

```shell
env FETCH_MEDIA=True MEDIA_TIMEOUT=500 ./archive ...
```

See [Configuration](/pirate/ArchiveBox/wiki/Configuration) page for more details about the available options and ways to pass config.  
If you're using Docker, also make sure to read the Configuration section on the [Docker](/pirate/ArchiveBox/wiki/Docker) page.

* * *

### [](#import-a-single-url-or-list-of-urls-via-stdin)Import a single URL or list of URLs via stdin

```shell
echo 'https://example.com' | ./archive
# or
cat urls_to_archive.txt | ./archive
```

You can also pipe in RSS, XML, Netscape, or any of the other supported import formats via stdin.

* * *

### [](#import-list-of-links-exported-from-browser-or-another-service)Import list of links exported from browser or another service

```shell
./archive ~/Downloads/browser_bookmarks_export.html
# or
./archive ~/Downloads/pinboard_bookmarks.json
# or
./archive ~/Downloads/other_links.txt
```

Passing a file as an argument here does not archive the file, it parses it as a list of URLs and archives the links _inside of it_, so only use it for _lists of links_ to archive, not HTML files or other content you want added directy to the archive.

* * *

### [](#import-list-of-urls-from-a-remote-rss-feed-or-file)Import list of URLs from a remote RSS feed or file

ArchiveBox will download the URL to a local file in `output/sources/` and attempt to autodetect the format and import any URLs found. Currently, Netscape HTML, JSON, RSS, and plain text links lists are supported.

```shell
./archive https://example.com/feed.rss
# or
./archive https://example.com/links.txt
```

Passing a URL as an argument here does not archive the specified URL, it downloads it and archives the links _inside_ of it, so only use it for RSS feeds or other _lists of links_ you want to add. To add an individual link use the instruction above and pass the URL via stdin instead of as an argument.

* * *

### [](#import-list-of-links-from-browser-history)Import list of links from browser history

```shell
./bin/archivebox-export-browser-history --chrome
./archive output/sources/chrome_history.json
# or
./bin/archivebox-export-browser-history --firefox
./archive output/sources/firefox_history.json
```

* * *

[](#ui-usage)UI Usage
---------------------

To access your archive, open `output/index.html` in a browser. You should see something [like this](https://archive.sweeting.me).

You can sort by column, search using the box in the upper right, and see the total number of links at the bottom.

Click the Favicon under the "Files" column to go to the details page for each link.

![](https://camo.githubusercontent.com/f0313b39a7005d3f761bc310bcaa3d680fe9461f/68747470733a2f2f692e696d6775722e636f6d2f3532526a68554d2e706e67)
 ![](https://camo.githubusercontent.com/30a722259232bea3021bc153a8e3bce3da9be32e/68747470733a2f2f692e696d6775722e636f6d2f476739735479712e706e67)

[](#disk-layout)Disk Layout
---------------------------

The `output/` folder containing the UI HTML and archived data has the structure outlined here.

```yaml
 - output/
   - index.json           # Main index of all archived URLs
   - index.html

   - archive/
      - 155243135/        # Archived links are stored in folders by timestamp
         - index.json     # Index/details page for individual archived link
         - index.html

         # Archive method outputs:
         - warc/          
         - media/
         - git/
         ...

   - sources/             # Each imported URL list is saved as a copy here
      - getpocket.com-1552432264.txt
      - stdin-1552291774.txt
      ...

   - static/              # Staticfiles for the archive UI
   - robots.txt
```

### [](#large-archives)Large Archives

I've found it takes about an hour to download 1000 articles, and they'll take up roughly 1GB.  
Those numbers are from running it single-threaded on my i5 machine with 50mbps down. YMMV.

Storage requirements go up immensely if you're using `FETCH_MEDIA=True` and are archiving many pages with audio & video.

You can run it in parallel by using the `resume` feature, or by manually splitting export.html into multiple files:

```shell
./archive export.html 1498800000 &  # second argument is timestamp to resume downloading from
./archive export.html 1498810000 &
./archive export.html 1498820000 &
./archive export.html 1498830000 &
```

Users have reported running it with 50k+ bookmarks with success (though it will take more RAM while running).

If you already imported a huge list of bookmarks and want to import only new bookmarks, you can use the `ONLY_NEW` environment variable. This is useful if you want to import a bookmark dump periodically and want to skip broken links which are already in the index.

[✏️ Help improve our documentation...](https://github.com/pirate/ArchiveBox/issues/new?assignees=&labels=&template=documentation_change.md&title=)

![](https://camo.githubusercontent.com/a8cb1ecef02f615c077eec5b7a6e11ffc9024ae9/68747470733a2f2f692e696d6775722e636f6d2f38793668765a612e706e67)

### Pages 15

*   **[Home](/pirate/ArchiveBox/wiki)**
*   **[Changelog](/pirate/ArchiveBox/wiki/Changelog)**
*   **[Chromium Install](/pirate/ArchiveBox/wiki/Chromium-Install)**
*   **[Configuration](/pirate/ArchiveBox/wiki/Configuration)**
*   **[Docker](/pirate/ArchiveBox/wiki/Docker)**
*   **[Donations](/pirate/ArchiveBox/wiki/Donations)**
*   **[Install](/pirate/ArchiveBox/wiki/Install)**
*   **[Publishing Your Archive](/pirate/ArchiveBox/wiki/Publishing-Your-Archive)**
*   **[Quickstart](/pirate/ArchiveBox/wiki/Quickstart)**
*   **[Roadmap](/pirate/ArchiveBox/wiki/Roadmap)**
*   **[Scheduled Archiving](/pirate/ArchiveBox/wiki/Scheduled-Archiving)**
*   **[Security Overview](/pirate/ArchiveBox/wiki/Security-Overview)**
*   **[Troubleshooting](/pirate/ArchiveBox/wiki/Troubleshooting)**
*   **[Usage](/pirate/ArchiveBox/wiki/Usage)**
*   **[Web Archiving Community](/pirate/ArchiveBox/wiki/Web-Archiving-Community)**

[![](https://camo.githubusercontent.com/d2e0fb21fdf1c6671deb25b1eb0a1181235e380a/68747470733a2f2f692e696d6775722e636f6d2f50564f3838415a2e706e67)
](Home)

[](#getting-started)Getting Started
===================================

*   [Quickstart](/pirate/ArchiveBox/wiki/Quickstart)
*   [Install](/pirate/ArchiveBox/wiki/Install)
*   [Docker](/pirate/ArchiveBox/wiki/Docker)

[](#documentation)Documentation
===============================

*   [Usage](/pirate/ArchiveBox/wiki/Usage)
*   [Configuration](/pirate/ArchiveBox/wiki/Configuration)
*   [Supported Sources](https://github.com/pirate/ArchiveBox#can-import-links-from-many-formats)
*   [Supported Outputs](https://github.com/pirate/ArchiveBox#saves-lots-of-useful-stuff-for-each-imported-link)
*   [Scheduled Archiving](/pirate/ArchiveBox/wiki/Scheduled-Archiving)
*   [Publishing Your Archive](/pirate/ArchiveBox/wiki/Publishing-Your-Archive)
*   [Chromium Install](/pirate/ArchiveBox/wiki/Chromium-Install)
*   [Security Overview](/pirate/ArchiveBox/wiki/Security-Overview)
*   [Troubleshooting](/pirate/ArchiveBox/wiki/Troubleshooting)

[](#more-info)More Info
=======================

*   [Roadmap](/pirate/ArchiveBox/wiki/Roadmap)
*   [Changelog](/pirate/ArchiveBox/wiki/Changelog)
*   [Donations](/pirate/ArchiveBox/wiki/Donations)
*   [Background & Motivation](https://github.com/pirate/ArchiveBox#background--motivation)
*   [Comparison to Other Tools](https://github.com/pirate/ArchiveBox#comparison-to-other-projects)
*   [Web Archiving Community](/pirate/ArchiveBox/wiki/Web-Archiving-Community)

* * *

[![](https://camo.githubusercontent.com/460e5fa9f0d33d4787aa85b4724d3c611e198a4e/68747470733a2f2f692e696d6775722e636f6d2f346e6b466a64762e706e67)
](https://archivebox.io)  
  
[![](https://camo.githubusercontent.com/83d4084f7b71558e33b08844da5c773a8657e271/68747470733a2f2f696d672e736869656c64732e696f2f747769747465722f75726c2f687474702f736869656c64732e696f2e7376673f7374796c653d736f6369616c)
](https://twitter.com/thesquashSH) [![](https://camo.githubusercontent.com/68fa8986092efeffee816dbd1c5a56d782eb703d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f446f6e6174652d50617472656f6e2d2532334444354437362e7376673f73616e6974697a653d74727565)
](https://www.patreon.com/theSquashSH)

##### Clone this wiki locally

  

*   © 2019 GitHub, Inc.
*   [Terms](https://github.com/site/terms)
*   [Privacy](https://github.com/site/privacy)
*   [Security](https://github.com/security)
*   [Status](https://githubstatus.com/)
*   [Help](https://help.github.com)

[](https://github.com "GitHub")

*   [Contact GitHub](https://github.com/contact)
*   [Pricing](https://github.com/pricing)
*   [API](https://developer.github.com)
*   [Training](https://training.github.com)
*   [Blog](https://github.blog)
*   [About](https://github.com/about)

You can’t perform that action at this time.

You signed in with another tab or window. Reload to refresh your session. You signed out in another tab or window. Reload to refresh your session.

![](blob:https://github.com/26c771ac-f4d9-472e-b690-372ad4b420b8)

Octotree

[](javascript:void(0))[](javascript:void(0))

Settings

GitHub access token

Hotkeys

 Open sidebar on hover

 Load entire tree at once

 Show in non-code pages

 Show file-specific icons

 Show pull request changes Note: maximum of 300 files 

Apply settings

Loving Octotree? You will love Octotree Pro even more. Octotree Pro is loaded with features that bring your productivity to the next level. [Learn more](https://www.octotree.io/?utm_source=lite&utm_medium=extension)

[Want this sidebar on the right?](https://www.octotree.io/?utm_source=lite&utm_medium=extension&utm_content=Want this sidebar on the right?)
